premodel.py

```diff
@@ -1,59 +1,61 @@
-# Libraries required to be imported
 import json
-import os
 import pickle
-import re
-import string
-import sys
+import pprint
 
-import fire
 import numpy as np
-from nltk.corpus import stopwords
-from nltk.tokenize import word_tokenize
-from sklearn.feature_extraction.text import TfidfVectorizer
 from sklearn.metrics.pairwise import linear_kernel
-from preprocess import preprocess_text, str_to_list
 
-
-def recommendation(query, tf, tfidf_matrix, idSummariesMap):
-    """ 
-        Takes the query, performs basic nlp processing, 
-        fits it into the tfidf vector space model, 
-        returns a list of the most similar papers using the cosine similarity on the model.
+from preprocess import preprocess_text
+
+
+def recommendation(query, tfidf_model, tfidf_matrix, paperTitles):
+    """
+        Takes the query, performs basic preprocessing. Converts it into a
+        tfidf vector. Then returns a list of paper titles most similar to
+        the query based on cosine similarity.
+
+        Parameters
+        ----------
+        query : string
+            Search query for the paper.
+        tfidf_model : sklearn.feature_extraction.text.TfidfVectorizer
+            TfidfVectorizer model
+        tfidf_matrix : np.array
+            Precomputed TF-IDF vectors of paper summaries.
+        paperTitles : list of string
+            List of titles of papers
      """
-    # process the query 
+    # Preprocess the query
     processedQuery = np.array(list(map(preprocess_text, query)))
-    
-    tfidfQuery = tf.fit_transform(processedQuery)
-    cosine_similarities = linear_kernel(tfidfQuery, tfidf_matrix)
-    related_docs_indices = cosine_similarities[0].argsort()[:-10:-1]
+
+    tfidfQuery = tfidf_model.transform(processedQuery)
+
+    cosine_similarities = linear_kernel(tfidf_matrix, tfidfQuery)
+    related_docs_indices = np.argsort(
+        cosine_similarities, axis=0)[-10:].reshape((-1, ))
```

Comment:
Since `related_docs_indices` has the top 10 papers. We need to reverse the args array.
```suggestion
        cosine_similarities, axis=0)[-10:].reshape((-1, ))[::-1]
```